orchestrator/src/test/java/com/scottlogic/deg/orchestrator/cucumber/features/operators/general/EqualTo.feature

```diff
@@ -1649,258 +269,20 @@ Feature: User can specify that a value is equalTo a required value
-      | foo  |
-      | null |
-
-  Scenario: 'EqualTo' run against a contradicting not 'afterOrAt' should only generate null
-    Given there is a field foo
-    And foo is equal to 2019-01-01T00:00:00.000Z
-    And foo is anything but after or at 2019-01-01T00:00:00.000Z
-    Then the following data should be generated:
-      | foo  |
-      | null |
-
-### before ###
-
-  Scenario: 'EqualTo' run against a non contradicting 'before' should be successful
-    Given there is a field foo
-    And foo is equal to 2019-01-01T00:00:00.000Z
-    And foo is before 2019-01-01T00:00:00.001Z
-    Then the following data should be generated:
-      | foo                      |
-      | null                     |
-      | 2019-01-01T00:00:00.000Z |
-
-  Scenario: 'EqualTo' run against a non contradicting not 'before' should be successful
-    Given there is a field foo
-    And foo is equal to 2019-01-01T00:00:00.000Z
-    And foo is anything but before 2019-01-01T00:00:00.000Z
-    Then the following data should be generated:
-      | foo                      |
-      | null                     |
       | 2019-01-01T00:00:00.000Z |
 
-  @ignore #594 As a user I expect values to be emitted in descending order
-  Scenario: Not 'equalTo' run against a non contradicting 'before' should be successful
-    Given there is a field foo
-    And foo is of type "datetime"
-    And foo is anything but equal to 2018-12-31T23:59:59.998Z
-    And foo is before 2019-01-01T00:00:00.000Z
-    And the generator can generate at most 5 rows
-    Then the following data should be generated:
-      | foo                      |
-      | 2018-12-31T23:59:59.999Z |
-      | 2018-12-31T23:59:59.997Z |
-      | 2018-12-31T23:59:59.996Z |
-      | 2018-12-31T23:59:59.995Z |
-      | 2018-12-31T23:59:59.994Z |
-
-  Scenario Outline: 'EqualTo' a non-datetime value with 'before' should be successful
-    Given there is a field foo
-    And foo is equal to <value>
-    And foo is before 2018-01-01T00:00:00.000Z
-    Then the following data should be generated:
-      | foo     |
-      | null    |
-      | <value> |
-    Examples:
-      | value |
-      | "a"   |
-      | 1     |
-
-  Scenario: 'EqualTo' run against a contradicting 'before' should only generate null
-    Given there is a field foo
-    And foo is equal to 2019-01-01T00:00:00.000Z
-    And foo is before 2019-01-01T00:00:00.000Z
-    Then the following data should be generated:
-      | foo  |
-      | null |
-
-  Scenario: 'EqualTo' run against a contradicting not 'before' should only generate null
-    Given there is a field foo
-    And foo is equal to 2019-01-01T00:00:00.000Z
-    And foo is anything but before 2019-01-01T00:00:00.001Z
-    Then the following data should be generated:
-      | foo  |
-      | null |
-
-### beforeOrAt ###
-
-  Scenario: 'EqualTo' run against a non contradicting 'beforeOrAt' should be successful
+  Scenario: 'EqualTo' run against a contradicting 'before' should generate no data
```

Comment:
Why test before & after but not before or at & after or at

Reply:
Also this is now under the comment section after?

Reply:
Probably just have a section called datetime constraints or something

Reply:
I think that would be a better way of organising constraint tests generally tbh, so we'd have a `### String Constraints ###` , `### Numeric Constraints ###` etc